aks/redis/resources.tf

```diff
@@ -0,0 +1,154 @@
+locals {
+  azure_name                  = "${var.resource_group_name}-${var.app_environment}-redis-${var.name}"
+  azure_private_endpoint_name = "${var.resource_group_name}-${var.app_environment}-redis-${var.name}-pe"
+}
+
+resource "azurerm_redis_cache" "main" {
+  count = var.use_azure ? 1 : 0
+
+  name                          = local.azure_name
+  location                      = data.azurerm_resource_group.main.location
+  resource_group_name           = data.azurerm_resource_group.main.name
+  capacity                      = var.azure_capacity
+  family                        = var.azure_family
+  sku_name                      = var.azure_sku_name
+  minimum_tls_version           = var.azure_minimum_tls_version
+  public_network_access_enabled = var.azure_public_network_access_enabled
+  redis_version                 = var.server_version
+
+  redis_configuration {
+    maxmemory_policy = "allkeys-lru"
```

Comment:
if we are using one redis module for all redis instances (some services have 2 redis caches), this would have to be passed to the module as it can differ between caches

Reply:
I've made this a configurable variable now, for services that need two Redis instances, they'd instantiate this module twice with a different name, etc.